### DISCUSSION TITLE
How can I let openjdk use system installed shared libraries ?

### DISCUSSION THREAD

**Discussion by JeremS** [Homebrew/discussions#4358] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 584588afb4ab557bcc0536f2794b4d043c333bb9
Last commit: 6 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 7e6d4c651682a6016309230ecf23d91497cf0c74
Core tap last commit: 5 weeks ago
Core tap branch: master
Core tap JSON: 26 Mar 14:47 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :1
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: dodeca-core 64-bit zen
Clang: N/A
Git: 2.40.0 => /home/linuxbrew/.linuxbrew/bin/git
Curl: 7.68.0 => /bin/curl
Kernel: Linux 5.15.0-67-generic x86_64 GNU/Linux
OS: Ubuntu 20.04.6 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 9.4.0
/usr/bin/ruby: N/A
glibc: 2.35_1
gcc@11: N/A
gcc: 12.2.0
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hi,

I am trying to use to use a java library called [JWM](https://github.com/HumbleUI/JWM). This library provides a native library (and jni bindings) that use X11 and OpenGL. When starting my app I get an error:
 `/tmp/skija_0.109.1_x64/libskija.so: libGL.so.1: cannot open shared object file: No such file or directory`

From what I understand it's because the openjdk installed by homebrew will only look for shared libs that homebrew installed itself. For instance If I `brew unlink openjdk` that app will start fine using the openjdk installed with apt. 

I tried `brew install mesa` but then `libXcursor.so.1` is missing. `brew install libxcursor` then it's `libGLX.so.0` that's missing...

Is there a way to tell the brew installed openjdk to look for shared libs in what's installed by the system (apt in my case)?

Cheers,

**Comment by carlocab** [comment]
CC @Homebrew/linux

**Reply by JeremS** [reply]
Hi,
I don't understand what you mean by 

> CC @Homebrew/linux

Sorry...

**Reply by iMichka** [reply]
@carlocab is asking help from the linux maintainer group with this message

**Comment by iMichka** [comment]
I think that use-case is not really supported. I think we did not get many requests for this.
In general we are not compatible with system libraries and do not allow mixing homebrew with system libraries.

The only solution I see is to recompile / reinstall `JWM` locally with our `openjdk`, or even better, build your own `JWM`formula.

**Reply by JeremS** [reply]
Building a formula or recompiling the native part or `JWM` kinda defeats the purpose of pulling a dependency via a `pom.xml`file when developing in java or a `deps.edn` file when developing in clojure (in my case).

Now I know homebrew doesn't use use libs provided by the host system except for a few exception. I suppose it's to keep a consistent environment and fair enough. It's just not really convenient for my use case :) 

I'll figure out how to manage my java stuff outside of then.

Cheers,
